### ISSUE TITLE
Error: database disk image is malformed

### ISSUE THREAD

**Issue by szsoftware** [greenrobot/greenDAO#741]
greenDao: 3.2.2

Even with a very simple Test entity (only @id(autoincrement=true)) I get a malformed sqlite3 database (tables not readable by Sqliteman or sqlite3 commandline).

greenDao can work with it but debuggung with external tools is impossible.

**Comment by greenrobot-team** [comment]
greenDAO just uses the Android SQLite APIs and standard SQL to create and access the database through the database open helper you pass to `DaoMaster`. You can verify the SQL by looking into the generated Dao classes (`createTable` method).
If the database is properly working on the device, this sounds more like an issue with the tools (sqlite3 or Sqliteman). -ut

**Comment by szsoftware** [comment]
Hi there,

you pointed me to a very strange thing.
On my formerly  used development phone I had no sqlite3 command on the device itself.
So I used:
sven@mylaptop$  adb -d shell "run-as mypackage cat /data/data/mypackage/databases/mydatabase.sqlite" > mydatabase.sqlite

I thought the db has transfered without any errors, because "file mydatabase" gives me correct header and my (mentioned) tools on the laptop see the tables and android metadata.

Now, after you pointed me to the tools, I took a cm-phone with sqlite3 on it.
There I can see the tables content as expected.

Quite late but now I did a "md5sum mydatabase.sqlite" on phone and laptop and, ohh wonder, they are different.

So, this is a way OT here, but just for reference: I must find out, why the transport corrupts the file that way, that it seems to be a (working) sqlite db at the end.

Anyways, thanks for your quick response and support so far.
Since I come from php and doctrine, it's really fun to work with your tools on Android!
